azure-pipelines.yml

```diff
@@ -36,15 +36,7 @@ jobs:
       architecture: $(PYTHON_ARCH)
   - script: python -m pip install --upgrade pip setuptools wheel
     displayName: 'Install tools'
-  - script: >-
-      python -m pip install
-      cython
-      matplotlib
-      numpy
-      pytest
-      pytest-xdist
-      scikit-learn
-      scipy
+  - script: python -m pip install -r maintainer\envs\minimal_requirements.txt
```

Comment:
This is also only the replacement of a single use case. If there were at least 3 places where the replacement was performed for `pip`, and where it did not change the meaning whatsoever, I might be in favor, but both the meaning (the set of deps installed) and the duplication requirement are not met for `pip`  at the moment.

Reply:
If you want to tell users to use that file, I think it is fine, but I'm currently at least -0.5 on using it in CI, at least until our CI grows further to use this more commonly, etc.

Reply:
Yes, I agree that `minimal_requirements.txt` is probably unnecessary. I wanted one of the CI suites to use `requirements.txt` and `environment.yml` to make sure they stayed up to date, and then for Travis it seemed easier to have a `minimal_environment.yml` that users could use than to write out packages again, so the `minimal_requirements.txt` was analogous. I'll remove it.